### PR TITLE
adds a `privateArtifacts` option to `compile.json`

### PR DESCRIPTION
**Pull request by johnspackman**
This adds a `privateArtifacts` option to `compile.json` to allow all urls to be locked within an application directory

### CODE REVIEW

**Review by cboulanger — APPROVED** [review]

File: docs/development/compiler/configuration/compile.md [hunk 1]
```diff
@@ -577,6 +577,29 @@ is the URI.
 It is up to you to implement the mapping inside your web server so that the
 "/some/virtual/uri/path/qooxdoo" URI is able to load the files from `../qooxdoo`
 
+### Keeping all files within the application directory
+
+When the compiler generates an application(s), it creates `transpiled` and `resource`
```
Comment (derrell) [inline docs/development/compiler/configuration/compile.md:582]: grammar problem with plural. How about "generates one or more applications..."

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by derrell — APPROVED** [review]

File: docs/development/compiler/configuration/compile.md (hunk 1 above)
Comment (johnspackman) [reply docs/development/compiler/configuration/compile.md]: done

**Review by hkollmann — APPROVED** [review]

**Comment by goldim** [comment]
@johnspackman I can't compile your PR bc the compiler complains on new option. Is it relative with `known-good`?

**Comment by johnspackman** [comment]
> Has this to be updated https://qooxdoo.org/schema/compile-1-0-0.json

Yes, I think that gets updated automatically when we do a release, but until then you'll get errors 

> after compiling with option nothing changed: all files on the same with app folder in target folder.

It is intentional that the files are still in the same place - but if you look at your generated `myappname/index.js` you'll see that the path to the files (resources and transpiled directories) has changed; the intention of this change is to allow you to have a URL on your web server where you only expose (or set permissions for) the application directory, and you would need to mount virtual folders within that URL.

EG in order to have an application at the root (eg `http://www.mywebsite.com/`, the `transpiled` and `resource` directory cannot be in `../transpiled`

Where there are multiple apps in a build, the `resource` and `transpiled` directories on disk need to be shared, hence the need to add virtual folders.

**Review by goldim — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**